### PR TITLE
[architecture/mvp] Boilerplate of architecture mvp

### PR DESCRIPTION
**Pull request by thanhnguyenduc157**
#### What's this PR do?

- [x] [FF-26] Boilerplate of architecture mvp

#### What are the relevant Git/YouTrack tickets?

- https://app.clickup.com/604513/606168/t/q4yn

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+}
+
+androidExtensions {
+    experimental = true
+}
+
+dependencies {
```
Comment (nambv) [inline app/build.gradle]: please add `dependencies.gradle` to manage app's dependencies @thanhnguyenduc157

Comment (nambv) [reply app/build.gradle]: 👍

Comment (thanhnguyenduc157) [reply app/build.gradle]: i added and use it

File: mvp/app/src/main/java/com/dwarvesv/mvp/view/list/ListPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        userRepository.getListData(GetUsersRequest("")).observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(
+                        {
+                            if (it.body() != null) {
+
+                                callHideIndicator(offset)
+
+                                it.body()?.let { it1 ->
```
Comment (nambv) [inline mvp/app/src/main/java/com/dwarvesv/mvp/view/list/ListPresenter.kt]: we should never use anonymous name with number like `it1`, `it2`. If more than one `it`, please provide a specify name to make the logic cleaner

**Comment by nambv** [comment]
@tanthanh289 @thanhnguyenduc157 please consider about implement this flow in `ListPresenter`:
1. Add database 
2. Fetch data from database first
3. Display data from database
3. Fetch data from the network
4. Save data from the network into database
5. Display latest data from the network

File: mvp/app/src/main/java/com/dwarvesv/mvp/view/list/ListPresenter.kt (hunk 2 above)
Comment (thanhnguyenduc157) [reply mvp/app/src/main/java/com/dwarvesv/mvp/view/list/ListPresenter.kt]: @nambv: agree, it will fix

**Pull request merged**